audio/filter/af_scaletempo2_internals.c

```diff
@@ -115,15 +119,36 @@ static void multi_channel_dot_product(
 {
     assert(frame_offset_a >= 0);
     assert(frame_offset_b >= 0);
-
-    memset(dot_product, 0, sizeof(*dot_product) * channels);
+#ifdef __aarch64__
     for (int k = 0; k < channels; ++k) {
+        float32x4_t dp = vmovq_n_f32(0.0);
+        const float* ch_a = a[k] + frame_offset_a;
+        const float* ch_b = b[k] + frame_offset_b;
+        int n = 0;
+        for (; n < num_frames - 3; n += 4) {
+            float32x4_t a_ld = vld1q_f32(ch_a);
+            float32x4_t b_ld = vld1q_f32(ch_b);
+            ch_a += 4;
+            ch_b += 4;
+            dp = vfmaq_f32(dp, a_ld, b_ld);
+        }
+        float after_simd = vaddvq_f32(dp);
+        for(; n < num_frames; n++) {
+            after_simd += *ch_a++ * *ch_b++;
+        }
+        dot_product[k] = after_simd;
+    }
+#else
+    for (; k < channels; ++k) {
         const float* ch_a = a[k] + frame_offset_a;
         const float* ch_b = b[k] + frame_offset_b;
+        float dp = 0;
         for (int n = 0; n < num_frames; ++n) {
-            dot_product[k] += *ch_a++ * *ch_b++;
+            dp += *ch_a++ * *ch_b++;
```

Comment:
also this change should still be a separate commit

Reply:
I'm just gonna do it the original way here because the compiler seems to optimise it to the same thing and everyone loves memset for some reason

Reply:
It doesn't tho. Just that it's apparently as fast.